src/EditorFeatures/CSharpTest/CodeActions/ReplacePropertyWithMethods/ReplacePropertyWithMethodsTests.cs

```diff
@@ -0,0 +1,388 @@
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAbstractProperty()
+        {
+            await TestAsync(
+@"class C {
+    public abstract int [||]Prop { get; } 
+    public void M() { var v = new { P = this.Prop } }
+}",
+@"class C {
+    public abstract int GetProp();
+    public void M() { var v = new { P = this.GetProp() } }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestVirtualProperty()
+        {
+            await TestAsync(
+@"class C {
+    public virtual int [||]Prop { get { return 1; } } 
+    public void M() { var v = new { P = this.Prop } }
+}",
+@"class C {
+    public virtual int GetProp() { return 1; }
+    public void M() { var v = new { P = this.GetProp() } }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestInterfaceProperty()
+        {
+            await TestAsync(
+@"interface I {
+    int [||]Prop { get; }
+}",
+@"interface I {
+    int GetProp();
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAutoProperty1()
+        {
+            await TestAsync(
+@"class C {
+    public int [||]Prop { get; }
+}",
+@"class C {
+    private readonly int prop;
+    public int GetProp() { return this.prop; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAutoProperty2()
+        {
+            await TestAsync(
+@"class C {
+    public int [||]Prop { get; }
+    public C() {
+        this.Prop++;
+    }
+}",
+@"class C {
+    private readonly int prop;
+    public int GetProp() { return this.prop; }
+    public C() {
+        this.prop = this.GetProp() + 1;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAutoProperty3()
+        {
+            await TestAsync(
+@"class C {
+    public int [||]Prop { get; }
+    public C() {
+        this.Prop *= x + y;
+    }
+}",
+@"class C {
+    private readonly int prop;
+    public int GetProp() { return this.prop; }
+    public C() {
+        this.prop = this.GetProp() * (x + y);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAutoProperty4()
+        {
+            await TestAsync(
+@"class C {
+    public int [||]Prop { get; } = 1;
+}",
+@"class C {
+    private readonly int prop = 1;
+    public int GetProp() { return this.prop; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplacePropertyWithMethods)]
+        public async Task TestAutoProperty5()
```

Comment:
Expression bodied properties, and properties with different access levels for their get and set methods?